textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -256,24 +273,50 @@ private ITuple joinTuples(ITuple outerTuple, ITuple innerTuple, JoinPredicate jo
 			return null;
 		}
 
-		// TODO Discuss and implement as to what Schema has to be.
-		// It shouldn't just be inner tuple or outer tuple schema.
-		Schema schema = innerTuple.getSchema();
-		List<IField> fieldList = innerTuple.getFields();
-		IField[] nextTupleField = new IField[fieldList.size()];
+		List<IField> newFieldList = new ArrayList<>();
+
+		ArrayList<Attribute> innerAttrList = new ArrayList<>();
+		innerAttrList.addAll(innerTuple.getSchema().getAttributes());
```

Comment:
"Moreover, operations involving schema should be done inopen(). If we do it in getNextTuple(), then there's a small overhead every time getNextTuple() is called." If we are certain that the schema will remain the same for all the tuples of an operator I will refactor the intersection of schemas to the `open()` method.